### DISCUSSION TITLE
expression highlight color

### DISCUSSION THREAD

**Discussion by ghost** [Qalculate/qalculate-gtk#470] [Q&A]
Does anyone know how to change the color of the highlight/background box that surrounds the entered expression when <enter>/'=' button clicked ?  I haven't changed any fonts/colors or themes etc.  The color is a dark blue and I did try the darkest shade if grey which sort of worked but when the box is gone the text is almost too light to see without it.  I've looked in the ~/.config/qalculate/ *.cfg files but did not seed anything I thought would help.

I'm using qalculate-gtk version 4.6.0 but had the same problem with 4.5.x. 

My os is fedora 36. I've tried using several desktops ( openbox, mate, cinnamon...) to no avail.  To be honest I don't remember having this issue before I updated from fedora 34.

I love this calculator and this is just a minor annoyance.
thanks

**Comment by hanna-kn** [comment]
Did you trying removing the "text_color=" line in ~/.config/qalculate/qalculate-gtk.cfg (specifying the text color will affect the readability of highlighted text with the default GTK theme)?

I do not know which GTK theme Fedora uses by default (if it is not Adwaita, you can try changing theme in the preferences of qalculate-gtk).

**Reply by ghost** [reply]
Hello hanna-kn, thank you for such a quick reply.

Yes, I did try changing the 'text_color=' line but my problem is with the color of the "box" which  surrounds the text when the '=' button is pressed.

![Screenshot from 2023-03-14 17-31-27](https://user-images.githubusercontent.com/33649535/225220072-3831b27f-c27e-461a-b8ba-92a9eb8d06ec.png)

My desktop manager theme is Adwaita and as you can see it is a light theme.  The color of the text is dark and so is the color of the "box".  This makes it difficult to see the text in the "box".  I changed the text to a lighter color but the setting in qalculate-gtk seems to be a global setting that changes the all the text.  This made the text (while in the "box") easy to read but then all the other text on the keypads, other widows, etc. was light text on light background. So now they were near impossible to read. 

Now I have the 'text_color' = to #000000 (black) which seems to work best for my sight.

thank you for a very nice qalculator :)

**Comment by hanna-kn** [comment]
> Yes, I did try changing the 'text_color=' line but my problem is with the color of the "box" which surrounds the text when the '=' button is pressed.

You should _remove_ the 'text_color=' line. This should happen automatically when the selected text color is identical to the default text color, which normally is black for Adwaita.

I assume that the issue occurs for all selected text (e.g. selected menu items).
